hi-ml-cpath/src/health_cpath/preprocessing/loading.py

```diff
@@ -204,38 +214,124 @@ def __call__(self, data: Dict) -> Dict:
         return data
 
 
+class LoadMaskSubROId(MapTransform, RandomizableTransform, BaseLoadROId):
+    """Transform that loads a pathology slide cropped to a sub-region of the mask bounding box (ROI).
+    The masks are supposed to be in png format, with the background in black (0) and tissue sections in categorical
+    values (e.g., 128, 255) along the RGB channels. The sub ROI is sampled from the foreground by randomly selecting a
+    label from unique values in the mask if roi_label is None. Otherwise the sub ROI is sampled from the foreground of
+    the specified label.
+
+    Operates on dictionaries, replacing the file paths in `image_key` loaded array, in (C, H, W) format. Also adds the
+    following meta-data entries:
+    - `'location'` (tuple): top-right coordinates of the bounding box
+    - `'size'` (tuple): width and height of the bounding box
+    - `'level'` (int): chosen magnification level
+    - `'scale'` (float): corresponding scale, loaded from the file
+    """
+
+    def __init__(
+        self,
+        image_key: str = SlideKey.IMAGE,
+        mask_key: str = SlideKey.MASK,
+        wsi_mag_at_level0: float = 10.0,
+        mask_mag: float = 1.25,
+        roi_label: Optional[Sequence[int]] = None,
```

Comment:
Type annotations for roi_label and usage are not consistent

Reply:
Thanks for catching it, I used to have a list of roi_labels than switch it to a single label